src/particles/BeamParticleContainerInit.cpp

```diff
@@ -59,7 +59,7 @@ namespace
         arrdata[BeamIdx::ux  ][ip] = ux * speed_of_light;
         arrdata[BeamIdx::uy  ][ip] = uy * speed_of_light;
         arrdata[BeamIdx::uz  ][ip] = uz * speed_of_light;
-        arrdata[BeamIdx::w][ip] = weight;
+        arrdata[BeamIdx::w][ip] = std::abs(weight);
```

Comment:
I would clip instead of flip signs:
```suggestion
        arrdata[BeamIdx::w][ip] = weight > 0._prt ? weight : 0._prt;
```

Reply:
I just saw: the signature of this function should probably use `amrex::ParticleReal` args instead of `amrex::Real`? :)